src/App/Platforms/Android/Utilities/AndroidHelpers.cs

```diff
@@ -64,5 +65,26 @@ public static PendingIntentFlags AddPendingIntentMutabilityFlag(PendingIntentFla
 
             return pendingIntentFlags;
         }
+
+        public static void StartActivityAndCollapseFromTileService(this Intent intent, TileService service, bool isMutable = false)
+        {
+            //For Android 14+ We need to use PendingIntent instead of Intent directly. Older versions still need to use Intent.
+            if (Build.VERSION.SdkInt >= BuildVersionCodes.UpsideDownCake)
+            {
+                var pendingIntent = PendingIntent.GetActivity(
+                    service.ApplicationContext,
+                    0,
+                    intent,
+                    AddPendingIntentMutabilityFlag(PendingIntentFlags.UpdateCurrent, isMutable)
+                );
+                if (pendingIntent == null) { return; }
+
+                service.StartActivityAndCollapse(pendingIntent);
+            }
+            else
+            {
+                service.StartActivityAndCollapse(intent);
+            }
+        }
```

Comment:
🎨 Oh sorry didn't see that we needed to do the `StartActivityAndCollapse` inside, so could we change this extension method to be from the `TileService` instead so it's more related to the action? Also, I'd invert the if and return to avoid the `else` and let the old versions be the specific alternative scenario. Finally, I see you're always specifying `isMutable` on the calls so no need to have a default value here:
```suggestion
        public static void StartActivityAndCollapseFromTileService(this TileService service, Intent intent, bool isMutable)
        {
            //For Android 14+ We need to use PendingIntent instead of Intent directly. Older versions still need to use Intent.
            if (Build.VERSION.SdkInt < BuildVersionCodes.UpsideDownCake)
            {
                service.StartActivityAndCollapse(intent);
                return;
            }
            var pendingIntent = PendingIntent.GetActivity(
                service.ApplicationContext,
                0,
                intent,
                AddPendingIntentMutabilityFlag(PendingIntentFlags.UpdateCurrent, isMutable)
            );
            if (pendingIntent == null) 
            { 
                return; 
            }
            service.StartActivityAndCollapse(pendingIntent);
        }
```